core/modules/replica/IngestClient.h

```diff
@@ -100,8 +103,12 @@ class IngestClient : public std::enable_shared_from_this<IngestClient>  {
                        unsigned int chunk,
                        bool isOverlap,
                        std::string const& inputFilePath,
-                       ColumnSeparator columnSeparator=COMMA,
-                       std::string const& authKey=std::string());
+                       std::string const& authKey=std::string(),
+                       std::string const& fieldsTerminatedBy=csv::Dialect::defaultFieldsTerminatedBy,
+                       std::string const& fieldsEnclosedBy=csv::Dialect::defaultFieldsEnclosedBy,
+                       std::string const& fieldsEscapedBy=csv::Dialect::defaultFieldsEscapedBy,
+                       std::string const& linesTerminatedBy=csv::Dialect::defaultLinesTerminatedBy,
```

Comment:
Good suggestion! Thank you!

Reply:
Actually, I'm taking my words back! Now I remember why I decided not to do this earlier. If I don't want to parse these strings in the client code because it may have a different version than the server. The idea here is to simply pass whatever values are provided by a user to the ingest server and let the latter validate the input. The user input will be sent AS-IS in the Protobuf request to the server. Consider a scenario where the implementation of the Ingest system has been updated to allow more options in the values of the parameters while letting clients to use an existing version of the client-side tool. The `mysql` command is probably a good example here because the command is agnostic to a specific version of the MySQL servers.

Reply:
But on client-side version of the package these four parameter will still be consistent with the client-size version of `Dialect`, no?

Reply:
Yes, that's still true. These defaults are the MySQL defaults. What may change would be a range of other values supported by each. The current implementation may have some limited subset of those. MySQL documentation is a bit fuzzy and controversial on this subject. Or, perhaps, I didn't just understand everything when I was reading it.

Reply:
Yeah, I think I asked different kind of question. In any case if you expect that server side may have a wider set of allowable characters in these parameters it may be perfectly reasonable to tell clients to upgrade client-side version of the package. In general I think it's hard to imagine that client side and server side versions will be different in this respect.